README.md

```diff
@@ -22,12 +24,192 @@ If you have issues with NEI outside of the GTNH modpack you may report them in t
 * Cycle between Recipe, Utility, and Cheat mode by ctrl clicking on the Wrench Icon
 * GT5u Tools/Items and GT6 tools should now properly work with the Overlay Recipe Transfer
 
-## Other items of note:
+### Other items of note:
 
 * Remove TMI style
 * Removed inventory Load/Save state
 
-## Development
+## Adding support for GTNH NEI features to your mod
+
+### Help! My mod is broken in your version of NEI! How can I fix it?
+
+Our version of NEI changes the recipe UI to be a bit more similar to [JEI][JEI]. As such, some legacy recipe handlers that assume the specific UI layout of old NEI might break even after providing additional metadata. Changes to the UI include the following: the recipe screen no longer has a fixed height, but can stretch to show more recipes at once. Additionally, the screen is no longer vertically centered, but at a fixed offset from the top of the MC window. Furthermore, the recipe paging widget has been moved from the bottom to the top of the recipe screen.
+
+The following sections list strategies for fixing broken handlers. Remember, you are always welcome to an issue in our repository if you are stuck with a broken recipe handler.
+
+#### I cannot update my mod, because it is no longer maintained, and I'm not a developer
+
+Some mods are no longer maintained and cannot be updated due to licensing. For those we provide a last-ditch effort called the "height hack". This tries to emulate the old NEI recipe layout, specifically on the vertical axis where our version of NEI had the most significant changes. This should help with misaligned tooltips and click zones especially.
+
+The height hack can be enabled by adding the fully-qualified class name of the recipe handler to the `heighthackhandlers.cfg` config file. If you encounter a recipe handler that requires this measure please open an issue and let us know.
+
+#### I'm a developer working on my own mod or maintaining a forked mod
+
+Please consider adding support for our IMC messages or events to automatically register handler metadata for your mod. Alternatively, we can provide default metadata for your recipe handlers in `assets/nei/csv/handlers.csv`, let us know the parameters, or even better, open a pull request. The fields available in the new metadata are described in the following section.
```

Comment:
I prefer _not_ allowing handlers.csv for actively maintained mods, they should use IMC or the API.  handlers.csv is for unmaintained/legacy mods.